back/src/domains/convention/use-cases/SendEmailsWhenAgencyIsActivated.unit.test.ts

```diff
@@ -0,0 +1,123 @@
+import { AgencyDtoBuilder, expectPromiseToFailWith } from "shared";
+import {
+  ExpectSavedNotificationsAndEvents,
+  makeExpectSavedNotificationsAndEvents,
+} from "../../../utils/makeExpectSavedNotificationsAndEvents";
+import { makeSaveNotificationAndRelatedEvent } from "../../core/notifications/helpers/Notification";
+import { CustomTimeGateway } from "../../core/time-gateway/adapters/CustomTimeGateway";
+import { TimeGateway } from "../../core/time-gateway/ports/TimeGateway";
+import { InMemoryUowPerformer } from "../../core/unit-of-work/adapters/InMemoryUowPerformer";
+import {
+  InMemoryUnitOfWork,
+  createInMemoryUow,
+} from "../../core/unit-of-work/adapters/createInMemoryUow";
+import { UuidV4Generator } from "../../core/uuid-generator/adapters/UuidGeneratorImplementations";
+import { SendEmailsWhenAgencyIsActivated } from "./SendEmailsWhenAgencyIsActivated";
+
+const emailPresentInBoth = "in-both@mail.com";
+const counsellorEmails = ["councellor@email.com"];
+const validatorEmails = ["toto-valide@email.com"];
+
+const agency = new AgencyDtoBuilder()
+  .withCounsellorEmails([...counsellorEmails, emailPresentInBoth])
+  .withValidatorEmails([...validatorEmails, emailPresentInBoth])
+  .withName("just-activated-agency")
+  .withLogoUrl("https://agency-logo.com")
+  .build();
+
+const agencyWithRefersTo = new AgencyDtoBuilder()
+  .withId("accompanying-agency-id")
+  .withRefersToAgencyId(agency.id)
+  .withCounsellorEmails(["accompanying-councellor@email.com"])
+  .withValidatorEmails(agency.validatorEmails)
+  .withName("just-activated-accompanying-agency")
+  .withLogoUrl("https://accompanying-logo.com")
+  .build();
```

Comment:
pas fan de ce terme : accompanying
je te propose :
```suggestion
const agencyWithRefersTo = new AgencyDtoBuilder()
  .withId("id-of-agency-refering-to-other")
  .withRefersToAgencyId(agency.id)
  .withCounsellorEmails(["councellor@email.com"])
  .withValidatorEmails(agency.validatorEmails)
  .withName("just-activated-agency-refering-to-other-one")
  .withLogoUrl("https://agency-refering-logo.com")
  .build();
```